src/features/chain-connection/WalletConnection.tsx

```diff
@@ -1,32 +1,91 @@
-import { useAccount, useConnect, useEnsName } from "wagmi";
+import { useAccount, useEnsName, useDisconnect, useConnect, Connector } from "wagmi";
 import { Button } from "../common/Button";
-import { InjectedConnector } from "wagmi/connectors/injected";
 import { twJoin } from "tailwind-merge";
 import truncateEthAddress from "truncate-eth-address";
+import { VscChromeClose } from "react-icons/vsc";
+import { useEffect } from "react";
 
-// TODO: this component should actually open a modal instead of defaulting to
-// the injected connector
-
-export const WalletConnection = ({}: {}) => {
-  const { address, isConnected } = useAccount();
+const WalletConnection = () => {
+  const { address, isConnected, isConnecting } = useAccount();
   const { data: ensName } = useEnsName({ address });
+  const { connect, connectors, isLoading, pendingConnector } = useConnect()
+  const { disconnect } = useDisconnect()
+
+  const dialog = document.querySelector('dialog')
+
+  const connectorButtons = () => {
+    return connectors.map((connector: Connector) => {
+      return (
+        <button
+          className="flex flex-row items-center p-2 border-2 border-border-dark rounded"
+          disabled={!connector.ready}
+          onClick={() => connect({ connector })}
+          key={connector.name}
+        >
+          <img
+            src={`src/assets/${connector.id}-logo.svg`}
+            alt='MetaMask logo'
+            className="mr-4 w-8 h-8"
+          />
+          <span>
+            {connector.name}
+            {!connector.ready && ' (unsupported)'}
+            {isLoading &&
+              connector.id === pendingConnector?.id &&
+              ' (connecting)'}
+          </span>
+        </button>)
+    })
+  }
 
-  const { connect } = useConnect({
-    connector: new InjectedConnector(),
-  });
+  useEffect(() => {
+    if (isConnected || isConnecting) {
+      dialog?.close()
+    }
+  }, [isConnected, isConnecting, dialog])
+
+  useEffect(() => {
+    dialog && dialog.addEventListener("click", e => {
```

Comment:
Two things here:

1. Don't use a `useEffect` to add an event listener, just use `onClick` on the element.
2. I would favour [this approach](https://stackoverflow.com/a/76201152/1398581) to close the dialog. Basically you're more directly checking if the click has hit the dialog, or a div it contains. If it's the div then you don't close, otherwise you do. Just be careful to set `padding` and `border` to zero on the dialog element as per the comments.

Reply:
the useEffect hook wasn't necessary. I replaced it